cadc-vos/src/main/java/ca/nrc/cadc/vos/Transfer.java

```diff
@@ -91,76 +88,49 @@
      */
     public int version = VOS.VOSPACE_20;
 
-    private URI target;
+    private List<URI> targets;
     private Direction direction;
+
     private View view;
     private Long contentLength;
     private String remoteIP;
     private List<Protocol> protocols;
     private boolean keepBytes = true;
     private boolean quickTransfer = false;
 
-    protected Transfer() { }
-
-    /**
-     * Constructor for copying a node.
-     * @param target
-     * @param direction
-     * @param protocols
-     */
-    public Transfer(URI target, Direction direction, List<Protocol> protocols)
-    {
-        this.target = target;
-        this.direction = direction;
-        this.protocols = protocols;
-        if (this.protocols == null)
-            this.protocols = new ArrayList<Protocol>();
-    }
-
     /**
      * Constructor for internal vospace move and copy operations.
      *
-     * @param target
-     * @param destination
+     * @param target - URI of target of transfer
+     * @param destination - URI of destination of transfer
      * @param keepBytes true for copy, false for move
      */
-    public Transfer(URI target, URI destination, boolean keepBytes)
-    {
-        this.target = target;
+    public Transfer(URI target, URI destination, boolean keepBytes) {
+        this.targets = new ArrayList<URI>();
+        this.targets.add(target);
         this.direction = new Direction(destination.toASCIIString());
         this.keepBytes = keepBytes;
         this.protocols = new ArrayList<Protocol>();
     }
 
     /**
-     * Constructor for copying a node with a specific view.
-     *
-     * @param target
-     * @param direction
-     * @param view
-     * @param protocols
+     * Ctor for Transfer. Use this for setting up transfers using multiple targets.
+     * Caller must call getProtocols().add(...) and getTargets().add(...)
+     * @param direction - Direction of transfer (to or from VOSpace)
      */
-    public Transfer(URI target, Direction direction, View view, List<Protocol> protocols)
-    {
-        this.target = target;
+    public Transfer(Direction direction) {
         this.direction = direction;
```

Comment:
I don't like the idea of a contructor not having all required items.  The object is in an unusable state after this because at least one target is required.  Why not have a constructor that takes a list?

Reply:
Based on previous design discussions, we're going with the constructors as they are, and adding code to instantiate the lists for targets and protocols at variable declaration time.